app/adapters/telegram_adapter/inbound.rb

```diff
@@ -50,19 +50,20 @@ def consume(telegram_message)
       yield(@message) if block_given?
     end
 
+    private
+
     def avatar_url(contributor)
       return unless contributor.telegram_id
 
-      profile_photos = Telegram.bot.get_user_profile_photos(user_id: contributor.telegram_id, limit: 1).with_indifferent_access
+      bot = contributor.organization.telegram_bot
```

Comment:
This is currently not possibly with the built in `telegram:bot:set_webhook` rake task we get from the gem, but we can achieve this by modifying it for our own purposes. First, I updated the routes, like so:

```rb
Telegram.bots.each do |(name, _bot)|
  scope ':organization_id' do
    telegram_webhook Telegram::WebhookController, as: "#{name}_telegram_webhook"
  end
end
```

Then, I added a rake task like so:

```rb
namespace :telegram do
  desc 'Set webhooks, scoped by organization id on all configured telegram bots'
  task set_webhooks: :environment do
    routes = Rails.application.routes.url_helpers
    Telegram.bots.each do |name, bot|
      route_name = "#{name}_telegram_webhook"
      organization = Organization.find_by(telegram_bot_api_key: bot.token)
      return unless organization

      url = routes.send("#{route_name}_url", organization_id: organization.id)
      print "Setting webhook for #{name}..."
      bot.set_webhook(url: url)
    end
  end
end
```

Reply:
Sorry, I didn't see the

```rb
def organization
  Organization.find_by(telegram_bot_api_key: bot.token)
end
```
Much better! So, now we just need to pass it to the adapter to find the contributor by the organization they are messaging.

Reply:
I think it would be cleaner to have a `:telegram_bot_api_token` dynamic param and scope the orute into `:organization`. I would prefer this.

However, after looking into the gem I found several places where our code would break if we divert too much away from what the gem author had in mind. It's not very suitable for a Telegram configuration that is available at runtime.

I was thinking about contributing, too. But that's for another day when I understand the gem a little bit better.